subprojects/file-collections/src/main/java/org/gradle/api/internal/file/FileCollectionFactory.java

```diff
@@ -96,16 +96,38 @@
      * Creates a {@link FileCollection} with the given files as content.
      *
      * <p>The collection is live and resolves the files on each query.
+     *
+     * <p>The collection fails to resolve if it contains providers which are not present.
      */
     FileCollectionInternal resolving(String displayName, Object sources);
 
     /**
      * Creates a {@link FileCollection} with the given files as content.
      *
      * <p>The collection is live and resolves the files on each query.
+     *
+     * <p>The collection ignores providers which are not present.
+     */
+    FileCollectionInternal resolvingIfPresent(String displayName, Object sources);
+
+    /**
+     * Creates a {@link FileCollection} with the given files as content.
+     *
+     * <p>The collection is live and resolves the files on each query.
+     *
+     * <p>The collection fails to resolve if it contains providers which are not present.
      */
     FileCollectionInternal resolving(Object sources);
 
+    /**
+     * Creates a {@link FileCollection} with the given files as content.
+     *
+     * <p>The collection is live and resolves the files on each query.
+     *
+     * <p>The collection ignores providers which are not present.
+     */
+    FileCollectionInternal resolvingIfPresent(Object sources);
```

Comment:
For me `ifPresent` means that code will only be executed if there is a value, otherwise we'll have a no-op. These new methods don't do that, but instead selectively ignore absent values. Something like `resolvingLeniently()` would capture this behavior better for me.

Reply:
You are right. I renamed the methods to `resolvingLeniently()`.